receiver/statsdreceiver/protocol/statsd_parser.go

```diff
@@ -220,6 +221,29 @@ func parseMessageToMetric(line string) (statsDMetric, error) {
 		result.metricType = metricspb.MetricDescriptor_GAUGE_INT64
 	}
 
+	// add metric_type dimension for all metrics
+
+	if enableMetricType {
+		var metricType = ""
+		switch result.description.statsdMetricType {
+		case "g":
+			metricType = "gauge"
+		case "c":
+			metricType = "counter"
+		}
```

Comment:
nit: Why not add all of the statsd and dogstatsd types here? 

I know we only have counters and gauges implemented, but if you add all the types now it saves us from having to do the work later.

Reply:
The problem is: if we add the new type, the unit tests will not be able to cover these lines currently. So I think this part will be better to add later.

Reply:
I see